contracts/protocol/libraries/logic/LiquidationLogic.sol

```diff
@@ -367,375 +366,63 @@ library LiquidationLogic {
-        ];
-        uint256 availableLiquidationAmount = vars.actualLiquidationAmount -
-            vars.liquidationProtocolFeeAmount;
-
-        if (availableLiquidationAmount > vars.actualDebtToLiquidate) {
-            SupplyLogic.executeSupply(
-                reservesData,
-                userConfig,
-                DataTypes.ExecuteSupplyParams({
-                    asset: params.liquidationAsset,
-                    amount: availableLiquidationAmount -
-                        vars.actualDebtToLiquidate,
-                    onBehalfOf: params.user,
-                    referralCode: 0
-                })
+            emit ReserveUsedAsCollateralEnabled(
+                params.liquidationAsset,
+                params.user
             );
-            if (!userConfig.isUsingAsCollateral(liquidationAssetReserve.id)) {
-                userConfig.setUsingAsCollateral(
-                    liquidationAssetReserve.id,
-                    true
-                );
-                emit ReserveUsedAsCollateralEnabled(
-                    params.liquidationAsset,
-                    params.user
-                );
-            }
-        } else {
-            vars.actualDebtToLiquidate = availableLiquidationAmount;
-        }
-
-        if (
-            vars.isLiquidationAssetBorrowed &&
-            vars.userTotalDebt == vars.actualDebtToLiquidate
-        ) {
-            userConfig.setBorrowing(liquidationAssetReserve.id, false);
         }
 
         // If the collateral being liquidated is equal to the user balance,
         // we set the currency as not being used as collateral anymore
-        if (vars.userCollateralBalance == 1) {
+        if (vars.userCollateral == 1) {
             userConfig.setUsingAsCollateral(collateralReserve.id, false);
             emit ReserveUsedAsCollateralDisabled(
                 params.collateralAsset,
                 params.user
             );
         }
 
-        if (vars.auctionEnabled) {
-            IAuctionableERC721(collateralReserve.xTokenAddress).endAuction(
-                params.collateralTokenId
-            );
-            emit AuctionEnded(
-                params.user,
-                params.collateralAsset,
-                params.collateralTokenId
-            );
-        }
-    }
-
-    function _executeInteractionsOfERC721LiquidationCall(
-        mapping(address => DataTypes.ReserveData) storage reservesData,
-        DataTypes.ExecuteLiquidationCallParams memory params,
-        LiquidationCallLocalVars memory vars
-    ) internal {
-        DataTypes.ReserveData storage liquidationAssetReserve = reservesData[
-            params.liquidationAsset
-        ];
         // Transfer fee to treasury if it is non-zero
-        if (vars.liquidationProtocolFeeAmount != 0) {
+        if (vars.liquidationProtocolFee != 0) {
             IERC20(params.liquidationAsset).safeTransferFrom(
-                vars.liquidator,
+                params.liquidator,
```

Comment:
Logic-wise, it seems inconsistent that in ERC-20 case, borrower pays for the protocol fees (extra discount on their collateral), whereas in ERC-721 case, liquidator pays for them. This makes the calculations for actualLiquidationAmount behave differently which might not be a good idea. My personal opinion is that we should stick to the concept that fees as taken from borrower's collateral in the form of an extra discount of the price paid for the NFT. So liquidationProtocolFee in both cases penalise the borrower but not the liquidator.